addons/binding/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/handler/DSMRMeterHandler.java

```diff
@@ -0,0 +1,205 @@
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // No comments can be handled
+    }
+
+    /**
+     * Initializes a DSMR Meter
+     *
+     * This method will load the corresponding configuration
+     */
+    @Override
+    public void initialize() {
+        logger.debug("Initialize MeterHandler for Thing {}", getThing().getUID());
+
+        Configuration config = getThing().getConfiguration();
+        DSMRMeterConfiguration meterConfig = null;
+        DSMRMeterDescriptor meterDescriptor = null;
+
+        if (config != null) {
+            meterConfig = config.as(DSMRMeterConfiguration.class);
+        } else {
+            logger.warn("{} does not have a configuration", getThing());
+        }
+
+        if (meterConfig != null) {
+            DSMRMeterType meterType = null;
+
+            try {
+                meterType = DSMRMeterType.valueOf(getThing().getThingTypeUID().getId().toUpperCase());
+            } catch (Exception exception) {
+                logger.error("Invalid meterType", exception);
+            }
+
+            if (meterType != null) {
+                meterDescriptor = new DSMRMeterDescriptor(meterType, meterConfig.channel);
+            }
+        } else {
+            logger.warn("Invalid meter configuration for {}", getThing());
+        }
+        if (meterDescriptor != null) {
+            meter = new DSMRMeter(meterDescriptor, this);
+
+            // Initialize meter watchdog
+            meterWatchdog = scheduler.scheduleWithFixedDelay(new Runnable() {
+                @Override
+                public void run() {
+                    if (System.currentTimeMillis()
+                            - lastValuesReceivedTs > DSMRMeterConstants.METER_VALUES_RECEIVED_TIMEOUT) {
+                        if (!getThing().getStatus().equals(ThingStatus.OFFLINE)) {
+                            updateStatus(ThingStatus.OFFLINE);
+                        }
+                    }
+                }
+            }, DSMRMeterConstants.METER_VALUES_RECEIVED_TIMEOUT, DSMRMeterConstants.METER_VALUES_RECEIVED_TIMEOUT,
+                    TimeUnit.MILLISECONDS);
+
+            updateStatus(ThingStatus.OFFLINE);
+        } else {
+            logger.warn("{} could not be initialized. Delete this Thing if the problem persists.", getThing());
+            updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "This could not be initialized. Delete Thing if the problem persists.");
+        }
+    }
+
+    /**
+     * Disposes this Meter Handler
+     */
+    @Override
+    public void dispose() {
+        if (meterWatchdog != null) {
+            meterWatchdog.cancel(false);
+            meterWatchdog = null;
+        }
+    }
+
+    /**
+     * Remove the Meter Thing
+     */
+    @Override
+    public void handleRemoval() {
+        // Stop the timeout timer
+        if (meterWatchdog != null) {
```

Comment:
Better call dispose instead of duplicating
@kaikreuzer is this a common pattern and should it maybe be the responsibility of the framework?